builder/core/project.py

```diff
@@ -26,7 +26,7 @@ def looks_like_code(path):
     return False
 
 
-def _apply_value(obj, key, new_value, apply_before=True):
+def _apply_value(obj, key, new_value, apply_before=False):
```

Comment:
Note: applying before by default is arguably wrong. we eval config in arch -> host -> target -> compiler -> version order, and we need to apply them in the same order, ex. host sets cmake flag to on, but we want to disable that cmake flag for a certain compiler, then we need to make sure compiler config value comes after host config value (before it was always inserted at the start).

This is just a spot fix and we do need to address how cmake flags are handled in a more generic way